internal/services/managedredis/managed_redis_access_policy_assignment_resource.go

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package managedredis
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/redisenterprise/2025-07-01/databases"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/redisenterprise/2025-07-01/redisenterprise"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/locks"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type ManagedRedisAccessPolicyAssignmentResource struct{}
+
+var _ sdk.Resource = ManagedRedisAccessPolicyAssignmentResource{}
+
+type ManagedRedisAccessPolicyAssignmentResourceModel struct {
+	ManagedRedisID string `tfschema:"managed_redis_id"`
+	DatabaseName   string `tfschema:"database_name"`
+	ObjectID       string `tfschema:"object_id"`
+}
+
+func (r ManagedRedisAccessPolicyAssignmentResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"managed_redis_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: redisenterprise.ValidateRedisEnterpriseID,
+		},
+
+		"database_name": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ForceNew: true,
+			Default:  "default",
+		},
+
+		"object_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.IsUUID,
+		},
+	}
+}
+
+func (r ManagedRedisAccessPolicyAssignmentResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r ManagedRedisAccessPolicyAssignmentResource) ModelObject() interface{} {
+	return &ManagedRedisAccessPolicyAssignmentResourceModel{}
+}
+
+func (r ManagedRedisAccessPolicyAssignmentResource) ResourceType() string {
+	return "azurerm_managed_redis_access_policy_assignment"
+}
+
+func (r ManagedRedisAccessPolicyAssignmentResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return databases.ValidateAccessPolicyAssignmentID
+}
+
+func (r ManagedRedisAccessPolicyAssignmentResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model ManagedRedisAccessPolicyAssignmentResourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			client := metadata.Client.ManagedRedis.DatabaseClient
+
+			clusterId, err := redisenterprise.ParseRedisEnterpriseID(model.ManagedRedisID)
+			if err != nil {
+				return fmt.Errorf("parsing managed redis ID: %+v", err)
```

Comment:
Avoid duplicate error wrapping. If you see inside `redisenterprise.ParseRedisEnterpriseID` there is already sufficient error message returned from there. Please check in other places as well.